Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM python:3.8-slim
+
+# Install Python tools (git + pipenv)
+RUN apt-get update && apt-get install -y git
+RUN pip install pipenv
+
+# Install memory_profiler if this script is run with PROFILE_MEMORY flag
+ARG INSTALL_MEMORY_PROFILER="false"
+RUN if [ "$INSTALL_MEMORY_PROFILER" = "true" ]; then \
+        apt-get update && apt-get install -y gcc && \
+        pip install memory_profiler; \
+    fi
```

Comment:
Do we need still need this cc @IsaackMwenda. Unless we're really sure that we do, I'm hesitant to add it now

Reply:
We can clean up for now and add on demand.

Reply:
removed